packages/nx/src/daemon/server/file-watching/file-watcher-sockets.ts

```diff
@@ -0,0 +1,83 @@
+import { Socket } from 'net';
+import { ProjectGraphCache } from '../../../project-graph/nx-deps-cache';
+import { PromisedBasedQueue } from '../../../utils/promised-based-queue';
+import { handleResult } from '../server';
+import { getProjectsAndGlobalChanges } from './changed-projects';
+
+const queue = new PromisedBasedQueue();
+
+export let registeredFileWatcherSockets: {
+  socket: Socket;
+  config: {
+    watchProjects: string[] | 'all';
+    includeGlobalWorkspaceFiles: boolean;
+  };
+}[] = [];
+
+export function removeRegisteredFileWatcherSocket(socket: Socket) {
+  registeredFileWatcherSockets = registeredFileWatcherSockets.filter(
+    (watcher) => watcher.socket !== socket
+  );
+}
+
+export function hasRegisteredFileWatcherSockets() {
+  return registeredFileWatcherSockets.length > 0;
+}
+
+export async function notifyFileWatcherSockets(
```

Comment:
I'd remove "async". We invoke this function in the fire and forget fashion without awaiting. IntelliJ will actually complain that you invoke a function returning promise and not awaiting